plugins/inputs/bigbluebutton/bigbluebutton.go

```diff
@@ -0,0 +1,167 @@
+func (b *BigBlueButton) gatherMeetings(acc telegraf.Accumulator) error {
+	body, err := b.GetMeetings()
+	if err != nil {
+		return err
+	}
+
+	var response MeetingsResponse
+	xml.Unmarshal(body, &response)
+
+	participantCount := 0
+	listenerCount := 0
+	voiceParticipantCount := 0
+	videoCount := 0
+	activeRecording := 0
+
+	if response.MessageKey == NoMeetingsMessageKey {
+		b.sendMeetingsRecord(acc, b.meetingsRecord(0, 0, 0, 0, 0))
+		return nil
+	}
+
+	for i := 0; i < len(response.Meetings.Values); i++ {
+		meeting := response.Meetings.Values[i]
+		participantCount += meeting.ParticipantCount
+		listenerCount += meeting.ListenerCount
+		voiceParticipantCount += meeting.VoiceParticipantCount
+		videoCount += meeting.VideoCount
+		if meeting.Recording == "true" {
+			activeRecording++
+		}
+	}
+
+	b.sendMeetingsRecord(acc, b.meetingsRecord(participantCount, listenerCount, voiceParticipantCount, videoCount, activeRecording))
+	return nil
+}
+
+func (b *BigBlueButton) meetingsRecord(participantCount int, listenerCount int, voiceParticipantCount int, videoCount int, activeRecording int) map[string]interface{} {
+	record := make(map[string]interface{})
+	record["participant_count"] = participantCount
+	record["listener_count"] = listenerCount
+	record["voice_participant_count"] = voiceParticipantCount
+	record["video_count"] = videoCount
+	record["active_recording"] = activeRecording
+	return record
+}
+
+func (b *BigBlueButton) recordingsRecord(recordingsCount int, publishedCount int) map[string]interface{} {
+	record := make(map[string]interface{})
+	record["recordings_count"] = recordingsCount
+	record["published_recordings_count"] = publishedCount
+	return record
+}
+
+func (b *BigBlueButton) sendMeetingsRecord(acc telegraf.Accumulator, record map[string]interface{}) {
+	b.sendRecord(acc, "bigbluebutton_meetings", record)
+}
+
+func (b *BigBlueButton) sendRecordingsRecord(acc telegraf.Accumulator, record map[string]interface{}) {
+	b.sendRecord(acc, "bigbluebutton_recordings", record)
+}
+
+func (b *BigBlueButton) gatherRecordings(acc telegraf.Accumulator) error {
+	body, err := b.GetRecordings()
+	if err != nil {
+		return err
+	}
+
+	var response RecordingsResponse
+	xml.Unmarshal(body, &response)
+
+	if response.MessageKey == NoRecordingMessageKey {
+		b.sendRecordingsRecord(acc, b.recordingsRecord(0, 0))
+		return nil
+	}
+
+	recordingsCount := 0
+	publishedCount := 0
+
+	for i := 0; i < len(response.Recordings.Values); i++ {
+		recording := response.Recordings.Values[i]
+		recordingsCount++
+		if recording.Published {
+			publishedCount++
+		}
+	}
+
+	b.sendRecordingsRecord(acc, b.recordingsRecord(recordingsCount, publishedCount))
+
+	return nil
+}
+
+func (b *BigBlueButton) tags() map[string]string {
+	tags := make(map[string]string)
+	tags["server_name"] = b.ServerName
+	return tags
+}
+
+func (b *BigBlueButton) sendRecord(acc telegraf.Accumulator, name string, record map[string]interface{}) {
+	acc.AddFields(name, record, b.tags())
+}
```

Comment:
Please reduce the number of indirections. It is hard to follow if you have to navigate through three levels where each level is just a rename and reorder of parameters just to arrive at `acc.AddFields()`. You could call `acc.AddFields()` directly in e.g. `gatherMeetings()` instead of `gatherMeetings() --> sendRecordingsRecord() --> sendRecord() --> acc.AddFields()`.